### ISSUE TITLE
[comp:popover] body 进行 scale 缩放之后popover如果处在页面的右下角，浮沉不会正常展示到左上角

### ISSUE THREAD

**Issue by clfeng** [IDuxFE/idux#1325]
- [x] I have searched the [issues](https://github.com/IDuxFE/idux/issues) of this repository and believe that this is not a duplicate.
          
### Reproduction link
[https://github.com/clfeng/idux-scale-bug-demo](https://github.com/clfeng/idux-scale-bug-demo)

### Steps to reproduce
1. 拉取项目
clone https://github.com/clfeng/idux-scale-bug-demo 项目

2. 安装依赖和运行项目
yarn install
yarn dev;

3. 提供了两个页面
http://localhost:5173/#/origin 页面为没有进行body缩放的表现行为

http://localhost:5173/#/scale 页面为进行了缩放后的页面的表现行为

### What is expected?
body 进行 scale 缩放之后popover如果处在页面的右下角，其表现行为和未进行缩放的表现一致，可计算得到底部不够空间进而展示在顶部

### What is actually happening?
body 进行 scale 缩放之后popover如果处在页面的右下角，浮沉不会正常展示到左上角，而是继续在下部显示，从而撑开了页面

### Environment Info 

```shell
Browsers:
    Chrome: 107.0.5304.121
    Firefox: 77.0.1
    Safari: 14.1.2
  npmPackages:
    @idux/cdk: ^1.2.1 => 1.2.1 
    @idux/components: ^1.2.1 => 1.2.1 
    @idux/pro: ^1.2.1 => 1.2.1 
    vue: ^3.2.41 => 3.2.45 
```

---
可参考 https://github.com/clfeng/idux-scale-bug-demo 项目READEM.md 的相关截图；
另外window下也可以直接复现问题

**Comment by clfeng** [comment]
初步定位跟 @floating-ui 库的实现有关系
![image](https://user-images.githubusercontent.com/20736207/205266353-da31d445-4e4d-439d-875e-2310ee341f2b.png)

**Comment by atomiks** [comment]
Hi @clfeng can you make an issue at Floating UI's [repo](https://github.com/floating-ui/floating-ui)?

I think it may be a repro for this issue no one has managed to make yet: https://github.com/floating-ui/floating-ui/pull/1763

**Comment by clfeng** [comment]
> Hi @clfeng can you make an issue at Floating UI's [repo](https://github.com/floating-ui/floating-ui)?
> 
> I think it may be a repro for this issue no one has managed to make yet: [floating-ui/floating-ui#1763](https://github.com/floating-ui/floating-ui/pull/1763)

I provide a demo based on float-ui to reproduce the bug and make an issue at Floating UI's too;
https://github.com/floating-ui/floating-ui/issues/2012

**Comment by danranVm** [comment]
closed #1362
